source/opt/type_manager.h

```diff
@@ -62,20 +62,20 @@ class TypeManager {
   // Returns the number of forward pointer types hold in this manager.
   size_t NumForwardPointers() const { return forward_pointers_.size(); }
 
- private:
-  using TypeToIdMap = std::unordered_map<const Type*, uint32_t>;
-  using ForwardPointerVector = std::vector<std::unique_ptr<ForwardPointer>>;
-
-  // Analyzes the types and decorations on types in the given |module|.
-  void AnalyzeTypes(const spvtools::ir::Module& module);
-
   // Creates and returns a type from the given SPIR-V |inst|. Returns nullptr if
   // the given instruction is not for defining a type.
   Type* RecordIfTypeDefinition(const spvtools::ir::Instruction& inst);
   // Attaches the decoration encoded in |inst| to a type. Does nothing if the
   // given instruction is not a decoration instruction or not decorating a type.
   void AttachIfTypeDecoration(const spvtools::ir::Instruction& inst);
 
+ private:
```

Comment:
PS: I have reverted this for now, and I will try to work around it in the client code.  However the motivation of making RecordIfTypeDefinition public was to enable adding of new types to the module.  As it is now, the type manager will get initialized at load time, but it will not see any types which get added to the module later...